checker/src/org/checkerframework/checker/units/UnitsAnnotatedTypeFactory.java

```diff
@@ -447,20 +1067,36 @@ private AnnotationMirror useUnitsRelation(Tree.Kind kind, UnitsRelations ur,
             return res;
         }
 
+        /**
+         * Returns the least upper bound of the two annotated types through the
+         * type factory
+         *
+         * @param lht left type
+         * @param rht right type
+         * @return the least upper bound of the two annotated types
+         */
+        private AnnotationMirror getLUB(AnnotatedTypeMirror lht, AnnotatedTypeMirror rht) {
```

Comment:
What is this method used for? It seems rather dangerous to have a method that might ignore many of the annotations on the LHS or RHS and just use whatever the first annotations are.